<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="vehicle_id" default="default"/>
  <let name="sensor_model" value="aip_xx1"/>
  <let name="parent_frame" value="base_link"/>
  <let name="input_map_with_wall_topic_name" value="/map/pointcloud_map"/>
  <let name="input_map_without_wall_topic_name" value="/map/pointcloud_map_without_wall"/>
  <arg name="accumulate_frame_num" default="2"/>
  <arg name="fitness_score_threshold" default="0.001"/>
  <arg name="config_file_name" default="extrinsic_map_based_calib.yaml"/>
  <arg name="use_calibration_area_map" default="true"/>

  <!-- extrinsic_calibration_client -->
  <arg name="src_yaml" default="$(find-pkg-share individual_params)/config/$(var vehicle_id)/$(var sensor_model)/sensors_calibration.yaml"/>
  <arg name="dst_yaml" default="$(env HOME)/sensors_calibration.yaml"/>

  <node pkg="extrinsic_calibration_client" exec="extrinsic_calibration_client" name="extrinsic_calibration_client" output="screen">
    <param name="src_path" value="$(var src_yaml)"/>
    <param name="dst_path" value="$(var dst_yaml)"/>
  </node>

  <!-- extrinsic_calibration_manager -->
  <node pkg="extrinsic_calibration_manager" exec="extrinsic_calibration_manager" name="extrinsic_calibration_manager" output="screen">
    <param name="parent_frame" value="$(var parent_frame)"/>
    <param name="child_frames" value="
    [velodyne_rear_base_link]"/>
    <param name="fitness_score_threshold" value="$(var fitness_score_threshold)"/>
  </node>

  <group>
    <include file="$(find-pkg-share point_cloud_accumulator)/launch/point_cloud_accumulator.launch.xml">
      <arg name="ns" value="$(var parent_frame)/velodyne_rear_base_link"/>
      <arg name="input_sensor_topic_name" value="/sensing/lidar/rear/pointcloud_raw"/>
      <arg name="output_sensor_topic_name" value="/sensing/lidar/rear/pointcloud_raw/accumulated_pointcloud"/>
      <arg name="accumulate_frame_num" value="$(var accumulate_frame_num)"/>
    </include>
    <include file="$(find-pkg-share extrinsic_map_based_calibrator)/launch/calibrator.launch.xml">
      <arg name="ns" value="$(var parent_frame)/velodyne_rear_base_link"/>
      <arg name="parent_frame" value="$(var parent_frame)"/>
      <arg name="child_frame" value="velodyne_rear_base_link"/>
      <arg name="config_file_name" value="$(var config_file_name)"/>
      <arg name="input_map_with_wall_topic_name" value="$(var input_map_with_wall_topic_name)"/>
      <arg name="input_map_without_wall_topic_name" value="$(var input_map_without_wall_topic_name)"/>
      <arg name="input_source_topic_name" value="/sensing/lidar/rear/pointcloud_raw/accumulated_pointcloud"/>
      <arg name="use_calibration_area_map" value="$(var use_calibration_area_map)"/>
      <arg name="sensor_model" value="$(var sensor_model)"/>
    </include>
  </group>
</launch>
